src/main/java/com/google/sps/servlets/EditCardServlet.java

```diff
@@ -0,0 +1,133 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import com.google.gson.Gson;
+import com.google.sps.tool.ResponseHandler;
+import com.google.sps.data.Folder;
+import com.google.sps.tool.GoogleTranslationAPI;
+import java.util.Map;
+import java.util.List;
+
+@WebServlet("/editcard")
+public class EditCardServlet extends HttpServlet{
+  
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      String newLabels = request.getParameter("labels");
+      String newfromLang = request.getParameter("fromLang");
+      String newToLang = request.getParameter("toLang");
+      String cardKey = request.getParameter("cardKey");
+      String newRawText = request.getParameter("rawText");
+      String newTextTranslated = GoogleTranslationAPI.translateText(newRawText, newToLang);
+      String newBlobKey = getBlobKey(request);
+      updateCard(response, newLabels, newfromLang, newToLang, cardKey, newRawText, newTextTranslated, newBlobKey);
+    }
+  }
+
+  private void updateCard(
```

Comment:
This function and the `setCardWithRequestedEdits` function duplicate logic in Card. Please fix.